R/compat-obj-type.R

```diff
@@ -74,7 +75,14 @@ obj_type_friendly <- function(x, value = TRUE, length = FALSE) {
         typeof(x),
         logical = if (x) "`TRUE`" else "`FALSE`",
         integer = "an integer",
-        double = "a number",
+        double =
+          if (abs(x) == Inf) {
+            "an infinite number"
+          } else if (is.nan(x)) {
+            "`NaN`"
```

Comment:
I think you need a test for this because I'm not sure it works — `is.na(NaN)` is `TRUE` so if I'm reading the code correctly it'll be caught by the previous `if` block.

Reply:
Good catch.